### PR TITLE
Updated tooltip positioning for the refactor

### PR DESCRIPTION
**Pull request by billti**
This addresses https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/198 .  I had to delete the old pull request and create a new one on top of the refactor work.

### CODE REVIEW

File: Default.sublime-keymap [hunk 1]
```diff
@@ -137,6 +128,15 @@
         ]
     },    // In case when auto match is enabled, only format if not within {}
     {
+        "keys": [ "alt+,"],
```
Comment (billti) [inline Default.sublime-keymap]: I had to move this in order for the right keyboard shortcut to show in the quick panel.

File: libs/popup_manager.py [hunk 2]
```diff
@@ -76,15 +80,37 @@ def display(self):
...
-            self.current_view.update_popup(popup_text)
+
+        arg_region = self.current_view.get_regions('argSpan')[0]
+        location = arg_region.begin()  # Default to start of arg list
+
+        # If the cursor is not in the first line of the arg list, set the popup
+        # location to first non-whitespace, or EOL, of the current line
+        cursor_point = self.current_view.sel()[0].begin()
+        opening_line = self.current_view.line(arg_region.begin())
+        if(not opening_line.contains(cursor_point)):
```
Comment (zhengbli) [inline libs/popup_manager.py]: No need for the parenthesis

File: TypeScript.sublime-commands [hunk 3]
```diff
@@ -10,5 +10,6 @@
  { "caption" : "TypeScript: Rename", "command": "typescript_rename" },
  { "caption" : "TypeScript: PasteAndFormat", "command": "typescript_paste_and_format" },
  { "caption" : "TypeScript: FormatLine", "command": "typescript_format_line" },
+ { "caption" : "TypeScript: SignatureInfo", "command": "typescript_signature_popup" },
```
Comment (zhengbli) [inline TypeScript.sublime-commands]: Maybe put the ``SignatureInfo`` below the ``FormatBlock``? As all the formatting ones were grouped together

File: libs/popup_manager.py [hunk 4]
```diff
@@ -28,6 +28,9 @@ def __init__(self, proxy):
         self.signature_index = 0
         self.current_parameter = 0
 
+        # Used to avoid looping
+        self.suppress_onhidden = 0
+
     def queue_signature_popup(self, view):
         cursor = view.rowcol(view.sel()[0].begin())
         point = Location(cursor[0] + 1, cursor[1] + 1)
```
Comment (zhengbli) [inline libs/popup_manager.py]: you can call ``get_location_from_view(view)`` in the ``view_helper`` module

File: libs/popup_manager.py [hunk 5]
```diff
@@ -28,6 +28,9 @@ def __init__(self, proxy):
         self.signature_index = 0
         self.current_parameter = 0
 
+        # Used to avoid looping
+        self.suppress_onhidden = 0
```
Comment (zhengbli) [inline libs/popup_manager.py]: Do we use the ``suppress_onhidden`` for the condition of anything? Or is it just for book keeping?

**Comment by zhengbli** [comment]
:+1:

**Pull request merged**